contracts/fuses/ebisu/EbisuZapperBalanceFuse.sol

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.26;
+
+import {SafeCast} from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+import {IMarketBalanceFuse} from "../IMarketBalanceFuse.sol";
+import {IERC20Metadata} from "@openzeppelin/contracts/token/ERC20/extensions/IERC20Metadata.sol";
+import {PlasmaVaultConfigLib} from "../../libraries/PlasmaVaultConfigLib.sol";
+import {IporMath} from "../../libraries/math/IporMath.sol";
+import {IPriceOracleMiddleware} from "../../price_oracle/IPriceOracleMiddleware.sol";
+import {PlasmaVaultLib} from "../../libraries/PlasmaVaultLib.sol";
+import {FuseStorageLib} from "../../libraries/FuseStorageLib.sol";
+import {ITroveManager} from "./ext/ITroveManager.sol";
+import {ILeverageZapper} from "./ext/ILeverageZapper.sol";
+import {EbisuZapperSubstrateLib, EbisuZapperSubstrate, EbisuZapperSubstrateType} from "./lib/EbisuZapperSubstrateLib.sol";
+
+/**
+ * @title Fuse for Ebisu protocol responsible for calculating the balance of the Plasma Vault in Ebisu protocol based on preconfigured market substrates
+ * @dev Substrates in this fuse are the address registries of Ebisu protocol that are used in the Ebisu protocol for a given MARKET_ID
+ */
+contract EbisuZapperBalanceFuse is IMarketBalanceFuse {
+    using SafeCast for int256;
+    uint256 public immutable MARKET_ID;
+
+    constructor(uint256 marketId) {
+        MARKET_ID = marketId;
+    }
+
+    function balanceOf() external view returns (uint256) {
+        bytes32[] memory substrates = PlasmaVaultConfigLib.getMarketSubstrates(MARKET_ID);
+
+        FuseStorageLib.EbisuTroveIds storage troveData = FuseStorageLib.getEbisuTroveIds();
+
+        uint256 substratesNumber = substrates.length;
+
+        if (substratesNumber == 0) return 0;
+        address collToken;
+        uint256 collTokenPrice;
+        uint256 collTokenPriceDecimals;
+        uint256 entireCollValue;
+        IPriceOracleMiddleware priceOracleMiddleware = IPriceOracleMiddleware(
+            PlasmaVaultLib.getPriceOracleMiddleware()
+        );
+        ITroveManager troveManager;
+        uint256 troveId;
+        EbisuZapperSubstrate memory target;
+
+        for (uint256 i; i < substratesNumber; ++i) {
+            target = EbisuZapperSubstrateLib.bytes32ToSubstrate(substrates[i]);
```

Comment:
In a different way, you interpret the substrates here and differently in the fuses. 
Fave you checked correctly in the tests, 
whether the balances were calculated correctly, 
whether the balance of the vault behaves correctly during changes/opening positions,  
whether the market balance behaves correctly.